toolkit/mozapps/handling/ContentDispatchChooser.jsm

```diff
@@ -231,16 +231,46 @@ XPCOMUtils.defineLazyModuleGetters(this, {
   MigrationUtils: "resource:///modules/MigrationUtils.jsm",
 });
 
+const { toggleRecording } = ChromeUtils.import(
+  "resource://devtools/server/actors/replay/connection.js"
+);
+
 // [Replay] - Mapping of replay: URL scheme values to destinations. Can either
 // be a URL or a function which invokes arbitrary browser-chrome functionality
 const replaySchemeMap = {
   library: 'https://app.replay.io/',
   migrate: (uri, principal, browsingContext) => {
-    const win = browsingContext.topFrameElement.getTabBrowser().owerGlobal;
+    const win = browsingContext.topFrameElement.getTabBrowser().ownerGlobal;
     MigrationUtils.showMigrationWizard(win, [
       MigrationUtils.MIGRATION_ENTRYPOINT_UNKNOWN,
     ]);
   },
+  record: (url, principal, browsingContext) => {
+    const parts = new URLSearchParams(url.query);
+    const target = parts.get("url");
+    const newtab = parts.has("newtab") ? parts.get("newtab").toLowerCase() === "true" : false;
+
+    if (!target) return;
+
+    const browser = browsingContext.topFrameElement;
+    if (newtab) {
+      const tabbrowser = browser.getTabBrowser();
+      const currentTabIndex = tabbrowser.visibleTabs.indexOf(tabbrowser.selectedTab);
+      const tab = tabbrowser.addTab(
+        target,
+        { triggeringPrincipal: principal, index: currentTabIndex === -1 ? undefined : currentTabIndex + 1}
+      );
+      tabbrowser.selectedTab = tab;
+    } else {
+      browser.loadURI(target, {
+        triggeringPrincipal: principal
+      });
+    }
+
+    new Promise(resolve => setTimeout(resolve, 500)).then(() => {
```

Comment:
I went looking around for a better way to handle the race here.  I _think_ the following might work:

```
tabbrowser.addEventListener("DOMContentLoaded", () => { toggleRecording(...); }, true);
```

However, I'm r+-ing this because it works.  I'll test this change locally after I clear some other stuff off my plate.

Reply:
Thanks! I'm rebuilding now and will give that a try too

Reply:
@kannanvijayan - This doesn't fire unforunately.

@bhackett1024 - Any ideas on how to know when a tab is ready to be recorded?

Reply:
I'm not sure what you mean, in principle we can specify whether to record when creating a new tab.  When we start recording an existing tab we are replacing it with a new content process via the updateBrowserRemoteness call in connection.js

Reply:
We noticed when trying this patch that without some sort of delay before calling `toggleRecording`, the new URL wasn't being loaded. My hope was to find a trigger that ensured that the correct URL was set on the content process before telling it record.

Reply:
Does this only happen in the new tab case?  It seems like it would be cleanest to create the new tab in an already-recording state, but I don't think we do that anywhere currently so some new plumbing would be needed.  It might be better for now to just leave the timeout in and file a followup so we can clean this up post launch.

Reply:
I'll file a follow up. Thanks!